rest/tests/test_rest.py

```diff
@@ -0,0 +1,579 @@
+                         Term.objects.get(slug=new_term_slug).label)
+        # never created a duplicate
+        self.assertEqual(1, Term.objects.count())
+
+        # test missing repository
+        # term permission work not done
+        self.get_term(self.repo.slug, vocab.slug, "missing",
+                      expected_status=HTTP_404_NOT_FOUND,
+                      skip_options_head_test=True)
+        self.assertEqual(0, self.get_terms(self.repo.slug, "missing")['count'])
+
+        vocab2 = create_vocabulary(
+            self.repo.id,
+            "other name",
+            "description",
+            True,
+            Vocabulary.FREE_TAGGING,
+            1000
+        )
+
+        self.delete_term(self.repo.slug, vocab.slug, new_term_slug,
+                         expected_status=HTTP_405_METHOD_NOT_ALLOWED)
+
+        # make sure terms only show up under vocabulary they belong to
+        input_dict = {
+            "label": "term label 2",
+            "weight": 3000,
+        }
+
+        term1 = self.create_term(self.repo.slug, vocab.slug, input_dict)
+        term2 = self.create_term(self.repo.slug, vocab2.slug, input_dict)
+        self.create_term(self.repo.slug, vocab.slug,
+                         expected_status=HTTP_400_BAD_REQUEST)
+
+        terms = self.get_terms(self.repo.slug, vocab.slug)
+        self.assertEqual(2, terms['count'])
+        terms = self.get_terms(self.repo.slug, vocab2.slug)
+        self.assertEqual(1, terms['count'])
+
+        self.get_term(self.repo.slug, vocab.slug, term1['slug'])
+        self.get_term(self.repo.slug, vocab.slug, term2['slug'],
+                      expected_status=HTTP_404_NOT_FOUND,
+                      skip_options_head_test=True)
+        self.get_term(self.repo.slug, vocab2.slug, term1['slug'],
+                      expected_status=HTTP_404_NOT_FOUND,
+                      skip_options_head_test=True)
+        self.get_term(self.repo.slug, vocab2.slug, term2['slug'])
+
+        # as anonymous
+        self.logout()
+        self.get_term(self.repo.slug, vocab.slug, term1['slug'],
+                      expected_status=HTTP_403_FORBIDDEN)
+        self.get_term(self.repo.slug, vocab.slug, term2['slug'],
+                      expected_status=HTTP_403_FORBIDDEN)
+        self.get_term(self.repo.slug, vocab2.slug, term1['slug'],
+                      expected_status=HTTP_403_FORBIDDEN)
+        self.get_term(self.repo.slug, vocab2.slug, term2['slug'],
+                      expected_status=HTTP_403_FORBIDDEN)
+
+    def test_repository_pagination(self):
+        """Test pagination for collections"""
+
+        expected = [
+            self.create_repository(
+                {
+                    'name': "name{i}".format(i=i),
+                    "description": "description"
+                }
+            ) for i in range(40)]
+
+        repositories = self.get_repositories()
+
+        # 100 we created + self.repo
```

Comment:
Typo? `100` should be `40`.

Photo I actually took last week:
![code comments](https://aoeus.com/code_comments.jpg)